### PR TITLE
Add WC core release thread tasks to release checklist

### PR DESCRIPTION
**Pull request by Aljullu**
Add some extra steps to the release checklist related to the WC core release thread.

### CODE REVIEW

File: .github/release-initial-checklist.md [hunk 1]
```diff
@@ -113,6 +113,10 @@ This only needs to be done if this release is the last release of the feature pl
   * Verify and make any additional edits to the pull request description for things like: Changelog to be included with WooCommerce core, additional communication that might be needed elsewhere, additional marketing communication notes that may be needed etc.
   * After the checklist is complete and the testing is done, it will be up to the WooCommerce core team to approve and merge the pull request.
 * [ ] Make sure you join the `#woo-core-releases` Slack channel to represent Woo Blocks for the release of WooCommerce core this version is included in.
+* [ ] Search the release thread of the WooCommerce core version in WooCommerce P2 (example: p6q8Tx-2gl-p2).
+  * [ ] Subscribe to it so you are aware of any news/changes.
```
Comment (frontdevde) [inline .github/release-initial-checklist.md]:
```suggestion
  * [ ] Subscribe to it, so you are aware of any news/changes.
```

File: .github/release-initial-checklist.md [hunk 2]
```diff
@@ -113,6 +113,10 @@ This only needs to be done if this release is the last release of the feature pl
...
+  * [ ] Make sure you are listed as the _Blocks Package_ lead or add yourself if you aren't.
+  * [ ] If there are any changes that you think might need to be highlighted, add them under the _Feature Rollup_ comment. Keep in mind a change to highlight might be included in a previous version from the one you just released, given that not all of them are included in WC core. Ping the porters if you are unsure.
```
Comment (frontdevde) [inline .github/release-initial-checklist.md]:
```suggestion
  * [ ] Add changes you want to highlight under the _Feature Rollup_ comment. Please also look at changes from earlier versions that we didn't include in WC core yet. Ping the porters if you are unsure.
```

**Review by frontdevde — APPROVED** [review]
LGTM 👍

Minor & optional: added a couple of changes that Grammarly suggested (comma & active voice).

**Review by opr — APPROVED** [review]
@frontdevde suggested changes look great, I will pre-approve but I think you should add them :)

**Comment by Aljullu** [comment]
Thanks for the reviews @frontdevde @opr! I applied the suggested changes so merging.

**Pull request merged**